packages/radio-group/src/vaadin-radio-group.d.ts

```diff
@@ -17,144 +20,93 @@ export type RadioGroupInvalidChangedEvent = CustomEvent<{ value: boolean }>;
+ * `invalid`           | Set when the element is invalid           | :host
+ * `focused`           | Set when the element is focused           | :host
+ * `has-label`         | Set when the element has a label          | :host
+ * `has-value`         | Set when the element has a value          | :host
+ * `has-helper`        | Set when the element has helper text      | :host
+ * `has-error-message` | Set when the element has an error message | :host
  *
  * See [Styling Components](https://vaadin.com/docs/latest/ds/customization/styling-components) documentation.
  *
  * @fires {CustomEvent} invalid-changed - Fired when the `invalid` property changes.
  * @fires {CustomEvent} value-changed - Fired when the `value` property changes.
  */
-declare class RadioGroupElement extends ThemableMixin(DirMixin(HTMLElement)) {
+declare class RadioGroup extends FieldMixin(
+  FocusMixin(DisabledMixin(KeyboardMixin(DirMixin(ThemableMixin(HTMLElement)))))
+) {
   /**
-   * The current disabled state of the radio group. True if group and all internal radio buttons are disabled.
-   */
-  disabled: boolean | null | undefined;
-
-  /**
-   * This attribute indicates that the user cannot modify the value of the control.
-   */
-  readonly: boolean | null | undefined;
-
-  /**
-   * This property is set to true when the value is invalid.
-   */
-  invalid: boolean;
-
-  /**
-   * Specifies that the user must fill in a value.
-   */
-  required: boolean | null | undefined;
-
-  /**
-   * Error to show when the input value is invalid.
-   * @attr {string} error-message
-   */
-  errorMessage: string;
-
-  /**
-   * String used for the label element.
-   */
-  label: string;
-
-  /**
-   * String used for the helper text.
-   * @attr {string} helper-text
-   */
-  helperText: string | null;
-
-  /**
-   * Value of the radio group.
+   * The value of the radio group.
+   *
+   * @attr {value} readonly
    */
   value: string | null | undefined;
 
-  _setFocused(focused: boolean): void;
-
-  _selectIncButton(next: boolean, checkedRadioButton: RadioButton): void;
-
-  _selectButton(element: RadioButton, setFocusRing?: boolean): void;
-
-  _containsFocus(): boolean;
-
-  _hasEnabledButtons(): boolean;
-
-  _selectNextButton(element: RadioButton): void;
-
-  _selectPreviousButton(element: RadioButton): void;
-
-  _changeSelectedButton(button: RadioButton | null, fireChangeEvent?: boolean): void;
-
   /**
-   * Returns true if `value` is valid.
+   * When present, the user cannot modify the value of the radio group.
+   * The property works similarly to the `disabled` property.
+   * While the `disabled` property disables all the radio buttons inside the group,
+   * the `readonly` property disables only unchecked ones.
    *
-   * @returns True if the value is valid.
-   */
-  validate(): boolean;
-
-  /**
-   * Returns true if the current input value satisfies all constraints (if any)
+   * @attr {boolean} readonly
```

Comment:
Please remove `@attr` annotation, we don't need it for non-camel-case properties.

Reply:
Done.